app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,70 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[ show edit update destroy ]
```

Comment:
Great application for before_action! 
We can improve it by removing the edit and update.